csharp/src/Ice/OutgoingFrame.cs

```diff
@@ -185,62 +197,65 @@ public CompressionResult CompressPayload()
                     return CompressionResult.PayloadNotCompressible;
                 }
 
-                int start = _encapsulationStart.Segment;
+                int start = PayloadStart.Segment;
```

Comment:
The start variable declaration could be moved after the binary context "if" block.

Reply:
Fixed.